src/main/scala/subsystem/HasTiles.scala

```diff
@@ -42,41 +101,94 @@ trait HasTiles extends HasCoreMonitorBundles { this: BaseSubsystem =>
   val tileCeaseXbarNode = IntXbar(p)
   val tileCeaseSinkNode = IntSinkNode(IntSinkPortSimple())
   tileCeaseSinkNode := tileCeaseXbarNode
+}
+
+/** HasTiles adds a Config-urable sequence of tiles of any type
+  *   to the subsystem class into which it is mixed.
+  */
+trait HasTiles extends HasCoreMonitorBundles with DefaultTileContextType
+{ this: BaseSubsystem => // TODO: ideally this bound would be softened to Attachable
+  implicit val p: Parameters
+  val tileAttachParams: Seq[CanAttachTile] = p(TilesLocated(location))
+  val tiles: Seq[BaseTile] = tileAttachParams.map(_.instantiate(p)) // actual tile creation
```

Comment:
Is the expectation that `p(TilesLocated(location))` lists tiles by order of hartid? If so, it would be nice to check hartid sanity here. We've run into problems in the past where instantiating and connecting tiles out of hartid order results in a broken DTS.

Reply:
Currently that is the expectation, but now that you mention it, now that all subtypes are ostensibly in one list I would be comfortable using `sortBy` to ensure that property. Would you prefer that to expecting/`requiring`?

Reply:
Yes, that would be fine. A sanity check for hartid uniqueness would still be useful.